source/includes/connection-snippets/x509-snippets/csharp-connection.cs

```diff
@@ -0,0 +1,53 @@
+// begin x509 connection
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Net.Security;
+
+using MongoDB.Bson;
+using MongoDB.Driver;
+using MongoDB.Driver.Core;
+using System;
+using System.Threading.Tasks;
+
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+using System.Security.Permissions;
+
+namespace WorkingWithMongoDB
+{
+    public class Entity
```

Comment:
Delete this class. No need for an Entity class here (it's not used anywhere I can see) and you can always use the default BsonDocument class instead for simple operations.

Reply:
@nathan-contino-mongo can you send me a snippet of how you want me to replace this, I copied and pasted this snippet almost verbatim from chris's repo and don't have my enviroment set up for csharp, so I'm hesitant on altering code like this without being able to run it

Reply:
The entire snippet within: "public class Entity{...}" (including the class declaration and the curly braces) should be deleted. It isn't used and it isn't necessary :)

Reply:
Fixed